capella/setup/disable_semantic_browser_auto_refresh.py

```diff
@@ -18,85 +18,115 @@
+        "1",
+        "true",
+        "t",
+    ):
+        logger.info(
+            "Environment variable `CAPELLA_DISABLE_SEMANTIC_BROWSER_AUTO_REFRESH` is set to `%s`. "
+            "The semantic browser auto refresh will not be disabled. ",
+            disable_semantic_browser,
+        )
+        sys.exit(0)
+
     logger.debug(
-        "File not found: `%s`."
-        " Cannot disable auto-refresh of semantic browser.",
-        FILE_PATH,
+        "Identified that the environment variable "
+        "`CAPELLA_DISABLE_SEMANTIC_BROWSER_AUTO_REFRESH` is set to `1`. "
+        "We will disable the auto-refresh of the semantic browser."
     )
-    sys.exit(0)
 
 
-def main() -> None:
-    tree = etree.parse(FILE_PATH)
-    root = tree.getroot()
+def check_for_file_existence(file_path: pathlib.Path) -> None:
+    if not file_path.is_file():
+        logger.debug(
+            "File not found: `%s`. "
+            "Cannot disable auto-refresh of semantic browser.",
+            file_path,
+        )
+        sys.exit(0)
+
+
+def identify_semantic_browser_line_no(file_path: pathlib.Path) -> int:
+    root = etree.parse(file_path).getroot()
     logger.debug(
         "Searching for XPath expression: `%s` in the file `%s`",
         XPATH_EXPR,
-        FILE_PATH,
+        file_path,
     )
     hit_elements = root.xpath(XPATH_EXPR, namespaces=NS_MAP)
     if not hit_elements:
         logger.debug("No elements found. Exiting.")
         sys.exit(0)
-    # runtime lands here, when we found a setting that controls if the semantic
+
+    # Runtime lands here, when we found a setting that controls if the semantic
     # browser should auto-refresh or not
     persisted_state = hit_elements[0]
     parent_element = persisted_state.getparent()
-    if parent_element is None:
-        sys.exit(0)
-    if "elementId" not in parent_element.attrib:
+    if (
+        parent_element is None
+        or "elementId" not in parent_element.attrib
+        or "semanticbrowser" not in parent_element.attrib["elementId"].lower()
+    ):
         sys.exit(0)
-    if "semanticbrowser" not in parent_element.attrib["elementId"].lower():
-        sys.exit(0)
-    # get line number of the element we want to modify
+
     browser_autorefresh_line_no = persisted_state.sourceline
     logger.debug(
         "Found element at line number: `%d`", browser_autorefresh_line_no
     )
-    line_no = 0
-    old = new = None
-    for line in fileinput.input(FILE_PATH, inplace=True):
-        line_no += 1
-        if line_no == browser_autorefresh_line_no:
-            old = "listeningToWorkbenchPageSelectionEvents=&quot;1&quot;"
-            new = "listeningToWorkbenchPageSelectionEvents=&quot;0&quot;"
-            if old in line:
+    return browser_autorefresh_line_no
+
+
+def replace_content_of_line(
+    file_path: pathlib.Path, line_no: int, old: str, new: str
+) -> None:
+    with fileinput.input(file_path, inplace=True, backup=".bak") as file:
```

Comment:
> What happens with the backup up file when something goes wrong in the upcoming for-loop?

The original file will be moved to the backup location. Since we pass the `backup` argument, it will not be deleted and will stay there until the next run happens.

> will the original and possible kind if destroyed (edited half-way) file be automatically replaced by the backup file or is this something the code author must do manually?

Since the file iterates over the lines of the file, it would probably just exit and leave the original file as it is.

Reply:
When we create the backup file, we must make use of it.
I tested that and the original file may get fucked up during the loop run.
The algorithm is not making use of the backup file:

<img width="1183" alt="image" src="https://github.com/user-attachments/assets/6688adeb-1d11-40f4-870c-6f73384665f7">

Reply:
> we must make use of it

I don't think that we must make use of it. It's just a backup that we can restore manually in case something crashes. We discussed it privately and came to the conclusion that we don't want to handle automatic restoring of the backup.